<launch><!-- 
  <arg name="mesh_path" default="package://dvrk_vision/defaults/femur.stl" />
  <arg name="scale" default=".001" /> -->

<!--   <arg name="mesh_path" default="package://oct_15_demo/resources/HillOrgan.obj" />
  <arg name="secondary_mesh_path" default="package://oct_15_demo/resources/HillOrganAnnotated.obj" />
  <arg name="texture_path" default="package://oct_15_demo/resources/DiffuseHills.png"/>
  <arg name="scale" default="1"/> -->

  <!-- <arg name="mesh_path" default="package://oct_15_demo/resources/largeProstate.obj" />
  <arg name="secondary_mesh_path" default="package://oct_15_demo/resources/largeProstate.obj" />
  <arg name="texture_path" default="package://oct_15_demo/resources/largeProstate.png"/>
  <arg name="scale" default="1.0526" /> -->

  <arg name="mesh_path" default="package://oct_15_demo/resources/smallProstate.obj" />
  <arg name="secondary_mesh_path" default="package://oct_15_demo/resources/smallProstate.obj" />
  <arg name="texture_path" default="package://oct_15_demo/resources/smallProstate.png"/>
  <arg name="scale" default="1.0526" />

  <include file="$(find dvrk_vision)/launch/just_cams.launch">
    <arg name="world_frame" default="PSM2_psm_base_link"/>
    <arg name="transform_yaml" default="package://dvrk_vision/defaults/registration_params.yaml"/>
  </include>

  <node ns="stereo" pkg="nodelet" type="nodelet" name="disparity"
        args="load stereo_image_proc/disparity standalone_nodelet"
        output="screen">
    <param name="queue_size" value="5" />
    <param name="approximate_sync" value="true" />
  </node>

  <node name="dynamic_reconfigure_load" pkg="dynamic_reconfigure" type="dynparam" args="load /stereo/disparity $(find dvrk_vision)/defaults/stereo_processing.yaml" />

  <node ns="stereo" pkg="nodelet" type="nodelet" name="point_cloud"
        args="load dvrk_vision/point_cloud2 standalone_nodelet">
    <param name="queue_size" value="5" />
    <remap from="left/image_rect_color" to="left/image_rect"/>
    <remap from="right/image_rect_color" to="right/image_rect"/>
    <param name="approximate_sync" value="true" />
  </node>

  <node ns="stereo" name="main_gui" type="main_gui.py" pkg="oct_15_demo" output="screen" required="true">
    <param name="mesh_path" value="$(arg mesh_path)"/>
    <param name="secondary_mesh_path" value="$(arg secondary_mesh_path)"/>
    <param name="mesh_scale" value="$(arg scale)"/>
    <param name="texture_path" value="$(arg texture_path)"/>
  </node>

  <node ns="stereo" name="bingham_registration" type="registration_node" pkg="bingham_registration" output="screen">
    <param name="mesh_path" value="$(arg mesh_path)"/>
    <param name="mesh_scale" value="$(arg scale)"/>
    <remap from="point_cloud" to="points2"/>
  </node>

  <node name="force_sensor" type="force_sensor_driver.py" pkg="force_sensor_gateway" output="screen"/>

	<!--   <node ns="stereo" name="registration_gui" type="registration_gui.py" pkg="dvrk_vision" output="screen">
	<param name="mesh_path" value="$(arg path)"/>
	<param name="mesh_scale" value="$(arg scale)"/>
	</node> -->

	<!--   <node ns="stereo" pkg="nodelet" type="nodelet" name="disparity_viewer"
	        args="load dvrk_vision/disparity standalone_nodelet" output="screen">
	    <param name="autosize" value="true" />
	    <remap from="image" to="disparity"/>
	  </node> -->

	<!--   <node ns="stereo" name="stereo_processing" type="stereo_processing.py" pkg="dvrk_vision" output="screen"/> -->
</launch>